hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -1339,6 +1339,19 @@
   /** Configuration key for the algorithm used for creating jks key, a string */
   public static final String CRYPTO_KEY_ALGORITHM_CONF_KEY = "hbase.crypto.key.algorithm";
 
+  /**
+   * Configuration key for the hash algorithm used for generating key hash in encrypted HFiles.
+   * This is a MessageDigest algorithm identifier string, like "MD5", "SHA-256" or "SHA-384".
+   * (default: "MD5" for backward compatibility reasons)
+   */
+  public static final String CRYPTO_KEY_HASH_ALGORITHM_CONF_KEY = "hbase.crypto.key.hash.algorithm";
```

Comment:
Please put these somewhere other than HConstants per this note:

>   // NOTICE!!!! Please do not add a constants here, unless they are referenced by a lot of classes.

Reply:
thanks, I'll move these